src/network/handler.rs

```diff
@@ -87,7 +85,7 @@ pub trait Client {
     }
   }
 
-  fn read_to_buf(&mut self, buffer: &mut BytesMut, size: usize) -> ClientResult {
+  fn read_to_buf(&mut self, buffer: &mut BytesMut) -> ClientResult {
```

Comment:
if we read two messages in a row, not specifying the size here means when we try to read the size prefix of the next message (4 bytes), they might have been read by the previous call?

Reply:
If the message is not read in one shot, the client get the state [ClientState::Await(size)](https://github.com/Geal/proust/blob/d34a337e95e0c0cf6bd444c73b763ec759b25584/src/network/handler.rs#L260). So yes the size have been read by the previous call.